networkx/algorithms/distance_measures.py

```diff
@@ -387,8 +387,11 @@ def diameter(G, e=None, usebounds=False, weight=None):
     return max(e.values())
 
 
-@nx._dispatchable
-def harmonic_diameter(G, sp=None):
+import math
+import networkx as nx
+
+@nx._dispatchable(edge_attrs="weight")
+def harmonic_diameter(G, weight=None, sp=None):
```

Comment:
Generally speaking, switching the order of non-keyword-only arguments is a no-go as it breaks backward compatibility. However, in this case I believe `harmonic_diameter` is not yet included in a released version so it should be fine.

I'd vote to eliminate this problem going forward by enforcing keyword-only arguments:

```suggestion
def harmonic_diameter(G, *, weight=None, sp=None):
```